lib/spack/spack/container/images.py

```diff
@@ -43,7 +43,8 @@ def build_info(image, spack_version):
 
     # Try to check if we have a tag for this Spack version
     try:
-        build_tag = image_data['build_tags'][spack_version]
+        # Translate version from git to docker if necessary
+        build_tag = image_data['build_tags'].get(spack_version, spack_version)
```

Comment:
We can probably simplify the try/except here, since having it wouldn't catch errors from user input anymore. I think we can also remove `build_tag` completely from the JSON files and substitute it with:

```suggestion
        # Translate version from git to docker if necessary
        build_tag = 'latest' if spack_version == 'develop' else spack_version
```